fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -96,7 +98,7 @@ pub fn expand_internal_struct(prop: &Property) -> Result<TokenStream, Error> {
         }
     }
 
-    let name = ident(&extract_struct_name_from_abi_property(prop).to_class_case());
+    let name = ident(&extract_custom_type_name_from_abi_property(prop, "struct")?.to_class_case());
```

Comment:
How about making this second argument an enum instead of a string? e.g.:

```rust
extract_custom_type_name_from_abi_property(prop, CustomType::Struct)
```

Then, in the function, we can use [strum](https://docs.rs/strum/latest/strum/) (we use it all over the repo already) to match the expected type, which is an enum, that we convert to string and match against `type_field[0]` like you already do.

This prevents hard-to-track bugs like calling this function passing `(prop, "strut")` or something hah.

Reply:
Yes that's what I wanted to do but didn't know how you'd feel about me introducing a new type. Happy to make it cleaner!